internal/states/encryption/singleton.go

```diff
@@ -0,0 +1,46 @@
+package encryption
+
+import (
+	"sync"
+
+	"github.com/opentofu/opentofu/internal/logging"
+)
+
+// GetSingleton is a big bad SingleTroll. It has green skin, smells like sweat and coffee beans, and is a generally
+// mean fella. Still, the SingleTroll keeps the procedural code from picking a fight with the OOP code, so it can stay.
+// However, if you can, please remove it.
+//
+// On a more serious note, a large portion of the OpenTofu codebase is still procedural, which means there is no way to
+// properly inject the Encryption object and carry the information it holds across subsystem boundaries. In most cases
+// you should use this function to get a globally scoped copy of the Encryption object. However, for tests you should
+// use the New() function and hopefully, some time in the future, we can get rid of the singleton entirely.
+//
+// If you are writing tests and you have the ability to inject an interface, please consider using the New() function
+// instead of relying on the singleton. If you must rely on this singleton, make sure that tests are not running in
+// parallel and that you call ClearSingleton() in the cleanup function:
+//
+//	func test_something(t *testing.T) {
+//	    // Do not call this: t.Parallel()
+//	    t.Cleanup(ClearSingleton)
+//	    encryption := encryption.GetSingleton()
+//	}
+func GetSingleton() Encryption {
+	lock.Lock()
+	defer lock.Unlock()
+	if instance == nil {
+		instance = New(logging.HCLogger())
+	}
+	return instance
+}
```

Comment:
Do you think it's worth adding a TODO/FIXME for added visibility about removing this.

Reply:
I don't think we should litter the code with TODOs, it will be come unmanagable very quickly. If we get rid of globally scoped things, this will fall out automatically.